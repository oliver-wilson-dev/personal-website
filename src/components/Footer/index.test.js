import React from 'react';
import { shallow } from 'enzyme';

jest
  .useFakeTimers('modern')
  .setSystemTime(new Date('1996-05-07'));

jest.mock('../SocialMedia', () => {
  const SocialMedia = () => null;

  return SocialMedia;
});

const defaultProps = {
  learnMoreClicked: jest.fn()
};

const render = (props = {}) => {
  let Footer;
  jest.isolateModules(() => {
    Footer = require('.').default;
  });

  return shallow(<Footer {...defaultProps} {...props} />);
};

describe('<Footer/>', () => {
  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when the learn more button is clicked', () => {
    it('should call the learnMoreClicked prop', () => {
      const component = render();

      component.find('button').simulate('click');

      expect(defaultProps.learnMoreClicked).toHaveBeenCalled();
    });
  });
});
