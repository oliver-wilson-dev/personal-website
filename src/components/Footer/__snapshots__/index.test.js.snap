// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render correctly 1`] = `
<footer
  className="flexCenter container"
>
  <div
    className="textContainer"
  >
    <span
      className="footerDisclaimerText"
    >
      By using this site you agree to our cookie policy.
       
      <button
        className="buttonLink"
        onClick={[MockFunction]}
        type="button"
      >
        <span
          className="buttonLinkText"
        >
          Read the cookie policy here
        </span>
        <span>
          .
        </span>
      </button>
    </span>
    <span
      className="footerDisclaimerText"
    >
      Icons made by
       
      <a
        href="https://www.flaticon.com/authors/freepik"
        title="Freepik"
      >
        Freepik
      </a>
       
      from
       
      <a
        href="https://www.flaticon.com/"
        title="Flaticon"
      >
        www.flaticon.com
      </a>
    </span>
    <span
      className="footerDisclaimerText"
    >
      The code for this website can be found on
       
      <a
        href="https://github.com/oliver-wilson-dev/personal-website"
        rel="noopener noreferrer"
        target="_blank"
      >
        github
      </a>
      .
    </span>
    <span
      className="footerDisclaimerText"
    >
      Â©
       
      1996
       
      Oliver Wilson
    </span>
  </div>
  <SocialMedia />
</footer>
`;
